### DISCUSSION TITLE
CSI Provisioning

### DISCUSSION THREAD

**Discussion by wibed** [samba-in-kubernetes/samba-operator#238] [Q&A] [answered]
How is csi to be provisioned in accordance with third party storage solutions?

is it intended to work as a clustered sidecar provisioner?
https://kubernetes-csi.github.io/docs/external-provisioner.html

**Comment by phlogistonjohn** [comment, marked as answer]
Supporting CSI provisioning is something on our radar but hasn't made a lot of progress lately.

My personal vision for CSI looks something like this:
* We reuse the existing smb driver for attaching to volumes: https://github.com/kubernetes-csi/csi-driver-smb
* We add a custom CSI provisioner that creates SmbShares dynamically 
  *  A [project](https://github.com/samba-in-kubernetes/csi-provisioner-smb) exists for this but hasn't made much progress so far
* Some efficiency enhancements in the operator (I've been looking into this)
* The ability to deploy it all neatly

I encourage anyone who reads this and is interested feel free to discuss this further here or to consider contributing to the project(s).


That all said, you could probably manually combine some of the existing pieces to have something that works in the short run - but perhaps not as cohesively as the above plan. It would be interesting to see how that works out!
